### PR TITLE
Watch Face UI

### PR DESCRIPTION
**Pull request by sanketshah92**
- Adding Watchface project file 
- Designed watch face using samsung watch studio
- Added time & date text & app shortcut complication area.

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (EntyR) [inline .idea/vcs.xml:6]: please, add all .idea folder into gitignore

**Review by EntyR — CHANGES_REQUESTED** [review]

**Comment by EntyR** [comment]
It's not not possible to export watchfacestudio project into android studio as far as I know. Please specify watchface directly in your project, can check android studio new project templates for reference

**Comment by EntyR** [comment]
Also where's no need to add any complications for now

**Comment by sanketshah92** [comment]
@EntyR i have updated WatchFace Service, do we need to use canvas to adjust background and draw texts for time and date? Please look into my approach.  I have used round shape image as a background, So for square watch it may be different look and feel.

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+                    startActivity(appIntent)
+                }
+
+            }
+            invalidate()
+        }
+
+        override fun onDraw(canvas: Canvas, bounds: Rect) {
+
+            val bitmap =
```
Comment (EntyR) [inline app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: please move any declaration out from onDraw method also please make a check in the begining to verify if date value was changed

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+ * their project, you will need to set your Configurations to
+ * "Do not launch Activity" for both the Wear and/or Application modules. If you
+ * are unsure how to do this, please review the "Run Starter project" section
+ * in the Google Watch Face Code Lab:
+ * https://codelabs.developers.google.com/codelabs/watchface/index.html#0
+ */
+class MyWatchFace : CanvasWatchFaceService() {
+
+    companion object {
+        private val NORMAL_TYPEFACE = Typeface.create(Typeface.SANS_SERIF, Typeface.NORMAL)
```
Comment (EntyR) [inline app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: please use roboco Typeface instead

**Comment by EntyR** [comment]
> my approach. I have used round shape image as a background, So for square watch it may be different look and feel.

As for square devices no need to worry about them it's app only for round devices, still if you want it to display on square you can use gradient instead of loading image resources, something like this

`private val colors = intArrayOf(
                Color.BLACK,
                Color.BLACK,
                context.getColor(R.color.purple_primary),
                Color.BLACK,
            )
`
`
            val backgroundColor = SweepGradient(
                bounds.exactCenterX(),
                bounds.exactCenterY(),
                colors,
                null
            )`

`val matrix = Matrix()`
`matrix.preRotate(27f, bounds.exactCenterX(), bounds.exactCenterY())
            backgroundColor.setLocalMatrix(matrix)
            mainPaint.shader = backgroundColor`

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt [hunk 4]
```diff
@@ -184,42 +190,42 @@ class MyWatchFace : CanvasWatchFaceService() {
         }
 
         override fun onDraw(canvas: Canvas, bounds: Rect) {
+            val scaledBitmap =
+                Bitmap.createScaledBitmap(backgroundImageBitmap, canvas.width, canvas.height, true)
```
Comment (EntyR) [inline app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: Please move it out of onDraw method, can reed [this ](https://developer.android.com/training/custom-views/custom-drawing)for more explanation

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt [hunk 5]
```diff
@@ -184,42 +190,42 @@ class MyWatchFace : CanvasWatchFaceService() {
...
 
+            mYOffset = backgroundImageBitmap.height - resources.getDimension(R.dimen._100sdp)
+            mXOffset = (canvas.width / 4.99).toFloat()
 
-            val timeFormat = "hh:mm a"
-            val timeText = LocalTime.now().format(DateTimeFormatter.ofPattern(timeFormat))
-
-            val dateFormat = "MMM dd,yyyy"
+            //Fetch time on every draw event
+             val timeText = LocalTime.now().format(DateTimeFormatter.ofPattern(WATCHFACE_TIME_FORMAT))
```
Comment (EntyR) [inline app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: Same thing here

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt (hunk 4 above)
Comment (sanketshah92) [reply app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: done

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt (hunk 5 above)
Comment (sanketshah92) [reply app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: done

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt (hunk 3 above)
Comment (sanketshah92) [reply app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: done

File: app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt (hunk 2 above)
Comment (sanketshah92) [reply app/src/main/java/com/sanket/watchapplication/presentation/watchface/MyWatchFace.kt]: done

**Review by AlexeyKolosov — APPROVED** [review]

**Comment by AlexeyKolosov** [comment]
I can see a lot of changes, which are not related to Watch Face UI, it is bad, please in future separate pull requests. Let's merge this.

**Pull request merged**